cdap-security/src/main/java/io/cdap/cdap/security/authorization/SecurityMetricsService.java

```diff
@@ -67,8 +68,12 @@ public MetricsContext createEntityIdMetricsContext(EntityId entityId) {
    */
   static Map<String, String> createEntityIdMetricsTags(EntityId entityId) {
     Map<String, String> tags = new HashMap<>();
-    for (EntityId currEntityId : entityId.getHierarchy()) {
-      addTagsForEntityId(tags, currEntityId);
+    // Adding default namespace tag
+    tags.put(Constants.Metrics.Tag.NAMESPACE, NamespaceId.SYSTEM.getNamespace());
```

Comment:
As discussed offline, many `entityId` extends `NamespaceEntityId`, so in `SecurityMetricsService#addTagsForEntityId` we should check:

```
if (entityId instanceOf NamespaceEntityId) {
  tags.put(Constants.Metrics.Tag.NAMESPACE, ((NamespaceEntityId) entityId).getNamespace());
}
```